tests/modules/ser.rs

```diff
@@ -5,6 +5,7 @@ use bson::{from_bson, oid::ObjectId, to_bson, Bson, EncoderError, EncoderResult}
 use std::{collections::BTreeMap, u16, u32, u64, u8};
 
 #[test]
+#[allow(clippy::float_cmp)]
```

Comment:
Clippy complains when we directly compare floats and suggests checking if they're within `std::f{32,64}::EPSILON` of each other. Since we're not doing any arithmetic with the floats, this shouldn't be an issue, so I've whitelisted each function that does this.